autoload/multiple_cursors.vim

```diff
@@ -1211,20 +1211,26 @@ function! s:wait_for_user_input(mode)
   endif
 
   if s:from_mode ==# 'i' && has_key(g:multi_cursor_insert_maps, s:last_char())
-    let c = getchar(0)
-    let char_type = type(c)
     let poll_count = 0
-    while char_type == 0 && c == 0 && poll_count < &timeoutlen
-      sleep 1m
+    let char_mapping = ""
+    while poll_count < &timeoutlen
```

Comment:
I'm not sure this looks quite right to me... we used to loop until we get one usable char, and now this little loop will consume as many chars as it can as long as there's no mapping? Perhaps I'm misunderstanding how `maparg` works?

Reply:
@eapache that is right.
Once you hit a char that is in **multi_cursor_insert_maps**:
```while (counter<timeoutlen) {
  char += new_char
  if (char matches a mapping) {
    char = resolved_mapping
    break
  }
  counter++
}
```

If nothing matches, the chars will be printed after **timeoutlen**.
If something matches, the {lhs} mapping will be used ("resolved mapping") in order to make those kind of mapping work:
```vim
imap jj <esc>
```